v2/tools/generator/internal/astmodel/type_definition_test.go

```diff
@@ -33,8 +33,7 @@ func Test_MakeTypeDefinition_GivenValues_InitializesProperties(t *testing.T) {
 	g.Expect(objectDefinition.Name().Name()).To(Equal(name))
 	g.Expect(objectDefinition.Type()).To(Equal(objectType))
 
-	actualGroup, actualVersion, ok := objectDefinition.Name().PackageReference().TryGroupVersion()
-	g.Expect(ok).To(BeTrue())
+	actualGroup, actualVersion := objectDefinition.Name().InternalPackageReference().GroupVersion()
```

Comment:
minor: Not sure if this was changed in this PR or an earlier one and I just noticed it, but IMO it'd be OK to call this method `PackageReference()` and just have it return an `InternalPackageReference`, since all internal type names by definition would have to be internal anyway, having the extra clarification in the method name doesn't necessarily disambiguate anything.

Reply:
Can't use co-variance on interface implementations in Go. 

The `PackageReference()` method on the `TypeName` interface must return a `PackageReference`, so I had to add a different method to return an `InternalPackageReference`. 

We may be able to get rid of `TypeName.PackageReference()` later, I'm not sure.